src/pages/[marketplace]/deals.tsx

```diff
@@ -0,0 +1,159 @@
+import { toUIAmount, formatRatio, formatTimestamp } from "../../utils/format.utils";
+import { Tabs } from "@components/Tabs";
+import { TabPane } from "@components/TabPane";
+import { Table } from "@components/Table";
+import { Icon, IconName } from "@components/Icon";
+import { Slider } from "@components/Slider";
+import { Button } from "@components/Button";
+import Link from "next/link";
+import Big from "big.js";
+import { useLocales } from "../../hooks/useLocales";
+import { useStore } from "@state/useStore";
+import Layout from "@components/Layout";
+import { NextPageWithLayout } from "pages/_app";
+
+const dealsTableColumns = [
+	{
+		title: "Name",
+		icon: "stacked-column-down" as IconName,
+		dataIndex: "name",
+		key: "name",
+	},
+	{
+		title: "Amount",
+		icon: "coins-alt" as IconName,
+		dataIndex: "amount",
+		key: "amount",
+		render: (text) => <span className="font-medium text-lg">{text} USDC</span>,
+	},
+	{
+		title: "Date",
+		icon: "calendar" as IconName,
+		dataIndex: "date",
+		key: "date",
+		render: (text) => <span className="font-medium text-lg">{text}</span>,
+	},
+	{
+		title: "Paid",
+		dataIndex: "paid",
+		key: "paid",
+		render: (value: number) => <Slider value={value} fullLabel="Full" />,
+	},
+];
+
+const Deals: NextPageWithLayout = () => {
+	const router = useRouter();
+	const { marketplace } = router.query;
+	const locales = useLocales();
+	const client = useCredixClient();
+	const maybeFetchMarket = useStore((state) => state.maybeFetchMarket);
+
+	const market = useStore((state) => state.market);
+	const [isLoadingDeals, setIsLoadingDeals] = useState<boolean>(true);
+	const [activeDeals, setActiveDeals] = useState<Deal[]>([]);
+	const [endedDeals, setEndedDeals] = useState<Deal[]>([]);
+
+	useEffect(() => {
+		maybeFetchMarket(client, marketplace as string);
+	}, [client, maybeFetchMarket, marketplace]);
+
+	const dealRepaidRatio = (principal: Big, principalAmountRepaid: Big) => {
+		if (!principalAmountRepaid.toNumber()) {
+			return 0;
+		}
+
+		return formatRatio(new Ratio(principal.toNumber(), principalAmountRepaid.toNumber()));
+	};
+
+	const mapDeal = useCallback(
+		({ address, name, principal, goLiveAt, principalAmountRepaid }: Deal) => {
+			return {
+				key: address.toString(),
+				name: name,
+				amount: toUIAmount(principal).toNumber(),
+				date: formatTimestamp(goLiveAt, locales as string[]),
+				paid: dealRepaidRatio(principal, principalAmountRepaid),
+			};
+		},
+		[locales]
+	);
+
+	const getDeals = useCallback(async () => {
+		setIsLoadingDeals(true);
+		const deals = await market?.fetchDeals();
+
+		if (!deals) {
+			setIsLoadingDeals(false);
+			return;
+		}
+
+		const { activeDeals, endedDeals } = deals?.reduce(
+			(acc, deal) => {
+				switch (deal.status) {
+					case DealStatus.IN_PROGRESS:
+						acc.activeDeals.push(mapDeal(deal));
+					case DealStatus.CLOSED:
+						acc.endedDeals.push(mapDeal(deal));
```

Comment:
maybe good to add `break` statements after each case because we don't return anything. A dealstatus can't be both in progress and closed so it shouldn't do any harm but maybe clearer that way